tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/FailureAnalysis/DnsResolutionFailureClassifier.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Azure.Sdk.Tools.PipelineWitness.Entities.AzurePipelines;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Services.FailureAnalysis
+{
+    public class DnsResolutionFailureClassifier : IFailureClassifier
+    {
+        public DnsResolutionFailureClassifier(VssConnection vssConnection)
+        {
+            this.vssConnection = vssConnection;
+            buildClient = vssConnection.GetClient<BuildHttpClient>();
+        }
+
+        private VssConnection vssConnection;
+        private BuildHttpClient buildClient;
+
+        private bool IsDnsResolutionFailure(string line)
+        {
+            return (line.Contains("EAI_AGAIN") && line.Contains("getaddrinfo"))
```

Comment:
A few comments on the detection logic:

1. I think these should be case-insensitive.
2. I think "getaddrinfo EAI_AGAIN" is redundant with the existing check for ("EAI_AGAIN" && "getaddrinfo").
3. I would remove the trailing period from "no such host is known".
4. Do you think we can match ("EAI_AGAIN" || "getaddrinfo") instead of ("EAI_AGAIN" && "getaddrinfo")?  In a failing build, I think either of these should be sufficient to suspect a DNS erron, and I doubt the logs would contain either if it **wasn't** a DNS failure.

Reply:
We can always filter down in the future if we find the less specific query produces false positives

Reply:
Fair call. I'll tweak these.